docs/model/datamodel.md

```diff
@@ -0,0 +1,342 @@
+
+In this section you will learn how to design and configure your data models for use with Graphback and how they can be used for multiple datasources.
+
+### @model
+
+The `@model` annotation is at the core of the Graphback's model definition syntax. It tells Graphback if a type should be considered part of the data model i.e. a table in a relational database or a collection in MongoDB.
+
+It also tells Graphback about what queries and mutations it needs to generate for a type. For Graphback to autogenerate type definitions and resolvers, there must be at least one type annotated with `@model`.
+
+This will generate all the default API definitions and resolvers:
+
+```graphql
+"""
+@model
+"""
+type Comment {
+  id: ID!
+  text: String
+  description: String
+}
+```
+
+You can customise what gets generated per model by adding [arguments](#arguments):
+
+```graphql
+"""
+@model(
+  delete: false
+)
+"""
+type Comment {
+  id: ID!
+  text: String
+  description: String
+}
+```
+
+Graphback generates the following queries, mutations and subscriptions:
+
+```graphql
+type Mutation {
+  createComment(input: CreateCommentInput!): Comment!
+  updateComment(input: MutateCommentInput!): Comment!
+}
+
+type Query {
+  getComment(id: ID!): Comment
+  findComments(
+    filter: CommentFilter
+    page: PageRequest
+    orderBy: OrderByInput
+  ): CommentResultList!
+}
+
+type Subscription {
+  newComment(filter: CommentSubscriptionFilter): Comment!
+  updatedComment(filter: CommentSubscriptionFilter): Comment!
+}
+```
+
+#### Arguments
+
+| Argument | Description | Example |
+|-|-|-|
+| `create` | Optionally specifies whether a `create` mutation is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(create: false)` |
+| `delete` | Optionally specifies whether a `delete` mutation is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(delete: true)` |
+| `find` | Optionally specifies whether a `find` query is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(find: false)` |
+| `findOne` | Optionally specifies whether a `findOne` query is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(findOne: false)` |
+| `update` | Optionally specifies whether a `update` mutation is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(update: true)` |
+| `subCreate` | Optionally specifies whether a `new` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subCreate: false)` |
+| `subUpdate` | Optionally specifies whether an `updated` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subUpdate: false)` |
+| `subDelete` | Optionally specifies whether a `deleted` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subDelete: false)` |
+
+### @oneToMany
```

Comment:
Out of the context. you are using reference style here where I would expect to get little bit smooth intro in model and have reference for full raw format